src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapperUtil.java

```diff
@@ -237,37 +227,198 @@ public static Object deserializeStoredVector(BytesRef storedVector, VectorDataTy
      * @return expected vector length
      */
     public static int getExpectedVectorLength(final KNNVectorFieldType knnVectorFieldType) {
-        int expectedDimensions = knnVectorFieldType.getDimension();
-        if (isModelBasedIndex(expectedDimensions)) {
-            ModelMetadata modelMetadata = getModelMetadataForField(knnVectorFieldType);
-            expectedDimensions = modelMetadata.getDimension();
-        }
+        int expectedDimensions = knnVectorFieldType.getKnnMappingConfig()
+            .getDimension()
+            .orElseGet(
```

Comment:
Can we make knnMappingConfg read dimension from model?

Reply:
I moved get dimension to be non-optional